samples/azure-quantum/entanglement-swap/ES-quantinuum-qiskit.ipynb

```diff
@@ -0,0 +1,930 @@
+    }
+   ],
+   "source": [
+    "for i in range(2,8):\n",
+    "    print(f\"{i} parties uses {entanglement_swap_n(i).num_qubits} qubits\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the quantinuum H1-2 has 12 qubits, we will limit our number of parties to $\\leq 6$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Validating entanglement swap with 2 parties\n",
+      "Job Status: job has successfully run\n",
+      "API validation succeeded for 2 parties!\n",
+      "Validating entanglement swap with 3 parties\n",
+      "Job Status: job has successfully run\n",
+      "API validation succeeded for 3 parties!\n",
+      "Validating entanglement swap with 4 parties\n",
+      "Job Status: job has successfully run\n",
+      "API validation succeeded for 4 parties!\n",
+      "Validating entanglement swap with 5 parties\n",
+      "Job Status: job has successfully run\n",
+      "API validation succeeded for 5 parties!\n",
+      "Validating entanglement swap with 6 parties\n",
+      "Job Status: job has successfully run\n",
+      "API validation succeeded for 6 parties!\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(2, 7):\n",
+    "    job_name = f\"Validating entanglement swap with {i} parties\"\n",
+    "    print(job_name)\n",
+    "    qc = entanglement_swap_n(i) \n",
+    "    validator_backend = provider.get_backend(\"quantinuum.hqs-lt-s2-apival\")\n",
+    "    qc_t = transpile(qc, validator_backend)\n",
+    "    job = validator_backend.run(qc_t, job_name=job_name)\n",
+    "    job_id = job.id()\n",
+    "    job_monitor(job)\n",
+    "    result = job.result()\n",
+    "    if not result.success:\n",
+    "        raise Exception(f\"API validation failed for {i} parties\")\n",
+    "    print(f\"API validation succeeded for {i} parties!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we see all our circuits can be run correctly, let us now run them on simulators. We will name our circuits so that when submitted as jobs we will be able to identify them in the Azure Quantum job management to have insight into costs. **Please note that this sample makes use of paid services on Azure Quantum. The cost of running this sample with the provided parameters on Quantinuum in a free trial subscription is approximately 111EHQC. This quantity is only an approximate estimate and should not be used as a binding reference. The cost of the service might vary depending on your region, demand and other factors.** We have a write up of our results below if you want to save on the credits."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_results = dict()\n",
+    "\n",
+    "def get_fidelity(counts):\n",
```

Comment:
Is this duplicated with the above definition?

Reply:
It isn't the qiskit simulator and cloud jobs have slightly different output formats. One of them has a space between digits, the other one does not

Reply:
Ah, makes sense; would it be possible, then, to separate out the part that normalizes the two dicts and share the rest? Or is that not worth it for as small a function, then?

Reply:
I think that'll just make the code harder to read and the function itself is very straightforward